src/main/java/com/vi/migrationtool/keycloak/KeycloakService.java

```diff
@@ -0,0 +1,84 @@
+package com.vi.migrationtool.keycloak;
+
+import java.io.IOException;
+import lombok.RequiredArgsConstructor;
+import org.springframework.boot.configurationprocessor.json.JSONException;
+import org.springframework.boot.configurationprocessor.json.JSONObject;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.client.ClientHttpResponse;
+import org.springframework.stereotype.Service;
+import org.springframework.util.LinkedMultiValueMap;
+import org.springframework.util.MultiValueMap;
+import org.springframework.web.client.ResponseErrorHandler;
+import org.springframework.web.client.RestTemplate;
+
+@Service
+@RequiredArgsConstructor
+public class KeycloakService {
+
+  private final KeycloakConfig keycloakConfig;
+
+  public void createRole(String roleName) {
+    var httpHeaders = new HttpHeaders();
```

Comment:
could you please split this method to smaller methods, to reduce complexity and make code cleaner